numba/tests/test_dispatcher.py

```diff
@@ -1021,6 +1022,73 @@ def execute_with_input():
         err = execute_with_input()
         self.assertEqual(err.strip(), "cache hits = 1")
 
+    def test_user_set_cpu_name(self):
+        self.check_pycache(0)
+        mod = self.import_module()
+        mod.self_test()
+        cache_size = len(self.cache_contents())
+
+        mtimes = self.get_cache_mtimes()
+        # Change CPU name to generic
+        try:
+            os.environ['NUMBA_CPU_NAME'] = 'generic'
+            self.run_in_separate_process()
+        finally:
+            del os.environ['NUMBA_CPU_NAME']
+        self.assertNotEqual(self.get_cache_mtimes(), mtimes)
+        self.assertNotEqual(len(self.cache_contents()), cache_size)
```

Comment:
Should these assert `>`, mtime increases and the cache has got bigger as a result of running the test?